File: destroyer.ts
const areAllIntegers = (...args: number[]) => args.every(x => Number.isInteger(x))

const destroyer = (numbers: number[], target1: number, target2: number) => {
  if (!Array.isArray(numbers))
    throw new Error("First input must be a valid array of numbers")

  if (!areAllIntegers(target1, target2)) {
    throw new Error("Second and third input must be valid numbers")
  }

  return numbers.filter(x => x !== target1 && x !== target2)
}

console.log(destroyer([1, 2, 3, 1, 2, 3], 2, 3))

File: destroyer.py
def validate_numbers(numbers: list[int]):
    if not numbers:
        raise ValueError("numbers must be valid and not empty")

def destroyer(numbers: list[int], target1: int, target2: int):
    validate_numbers(numbers)
    # return [x for x in numbers if x != target1 and x != target2]
    # return list(x for x in numbers if x != target1 and x != target2)
    return list(filter(lambda x: x != target1 and x != target2, numbers))


print(destroyer([1, 2, 3, 1, 2, 3], 2, 3))  # [1, 1]
print(destroyer([1, 2, 3, 5, 1, 2, 3], 2, 3))  # [1, 5, 1]
